src/app/hooks/useCrawl.ts

```diff
@@ -2,18 +2,28 @@ import axios, { AxiosRequestConfig } from 'axios';
 import { useFormContext } from 'react-hook-form';
 import type { ContextFormValues } from '@/components/Context/types';
 
+interface Props {
+  url: string;
+  splittingMethod: string;
+  chunkSize: number;
+  overlap: number;
+  maxDepth: number;
+  maxPages: number;
+  namespace: string;
+}
+
 export const useCrawl = () => {
   const { setValue } = useFormContext<ContextFormValues>();
 
-  const crawl = async (
-    url: string,
-    splittingMethod: string,
-    chunkSize: number,
-    overlap: number,
-    namespace: string,
-    maxDepth: number,
-    maxPages: number
```

Comment:
@athrael-soju if we pass more than 2-3 args, let's pass it like object

Reply:
agreed